src/Symfony/Component/HttpKernel/Bundle/MicroBundle.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Bundle;
+
+use Symfony\Component\Config\Definition\Builder\NodeParentInterface;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface;
+use Symfony\Component\DependencyInjection\Extension\ExtensionInterface;
+
+/**
+ * A all-in-one implementation of {@link BundleInterface},
+ * {@link ExtensionInterface} and {@link ConfigurationInterface}.
+ *
+ * @author Guilhem N. <egetick@gmail.com>
+ */
+abstract class MicroBundle extends Bundle implements ExtensionInterface, ConfigurationExtensionInterface, ConfigurationInterface
```

Comment:
After all I don't think we should do that. `ConfigurationExtensionInterface` is used in commands such as `ConfigDebugCommand` and `ConfigDumpReferenceCommand` (should be at least, they won't work if the extension doesn't implement this interface).
And as `ConfigurationExtensionInterface` returns an instance of `ConfigurationInterface`, the `MicroBundle` must be an instance of `ConfigurationInterface`.
About `PrependExtensionInterface` people only need to implement it themself to the `MicroBundle`. I don't think it should be used by default as this is not useful most of the time and won't bring anything to the user.